### PR TITLE
Add php 7.1 support

### PR DESCRIPTION
**Pull request by jokesterfr**
- Bump dependencies
- Disable allure and target 8.2 by default
- Switch to alpine linux in dev-tools Dockerfile
- Support various version of xdebug in dev-tools
- Guzzle clients refactoring (Sync API & Collector API)

### CODE REVIEW

File: src/Api/CollectorApiClient.php [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+     * Push some ShopContents to CloudSync
+     *
+     * @param string $jobId
+     * @param string $data
+     * @param int $startTime in seconds since epoch
+     * @param bool $fullSyncRequested
+     *
+     * @return array
+     */
+    public function upload(string $jobId, string $data, int $startTime, bool $fullSyncRequested = false)
```
Comment (Clap404) [inline src/Api/CollectorApiClient.php:61]: si tu es chaud : upload et delete me semblent identique, à l'exception de l'url...

File: src/Api/CollectorApiClient.php [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+     * @return float
+     */
+    public function getRemainingTime(int $startTime)
+    {
+        $remainingTime = time() - $startTime;
+        if ($remainingTime <= 0) {
+            return 0;
+        }
+
+        return $remainingTime - Config::COLLECTOR_MAX_TIMEOUT - 1.5;
```
Comment (Mikatux) [inline src/Api/CollectorApiClient.php:154]: Pourquoi 1.5 🤔

**Review by Mikatux — APPROVED** [review]
C'est surtout du renaming ? 
Et ca semble OK

**Review by Clap404 — APPROVED** [review]
Belle refacto !

File: src/Api/CollectorApiClient.php (hunk 2 above)
Comment (jokesterfr) [reply src/Api/CollectorApiClient.php:154]: In the worst case the timeout will be fully consumed, and no more time will be left for the JSON processing and state storage in the MySQL database.

Comment (jokesterfr) [reply src/Api/CollectorApiClient.php:168]: Just added more comments on this part, let me know if this is anyway better like this :)

File: src/Api/CollectorApiClient.php (hunk 1 above)
Comment (jokesterfr) [reply src/Api/CollectorApiClient.php:72]: verb and endpoint differs, but you're right, can be slightly reduced

**Review by Quetzacoalt91 — APPROVED** [review]

**Comment by jokesterfr** [comment]
PR still pending, I could not get the expected behavior. Probably related to guzzle client usage.

**Pull request merged**